### PR TITLE
Honor silent flag & prevent error handler catch

### PR DESCRIPTION
**Pull request by mdarse**
This is a more generic fix to https://github.com/php-vcr/php-vcr/pull/35, it should fix https://github.com/php-vcr/php-vcr/issues/108 too.

### CODE REVIEW

File: src/VCR/Util/StreamProcessor.php [hunk 1]
```diff
@@ -280,22 +280,23 @@ public function stream_tell()
...
     {
         $this->restore();
-        try {
+        if ($flags & STREAM_URL_STAT_QUIET) {
+            set_error_handler(function() {
+                // Use native error handler
+                return false;
+            });
             $result = @stat($path);
-        } catch (\ErrorException $e) {
```
Comment (adri) [inline src/VCR/Util/StreamProcessor.php]: Looks nice thank you. I wonder though, if we don't catch the exception PHPUnit might throw ErrorExceptions when running in `processIsolation`. Did you test this by any chance?

**Comment by mcrumm** [comment]
:+1: for this.  I'm using [kahlan](https://github.com/crysalead/kahlan) for specs, and ran into #33 when I added a dependency that referenced an existing namespace. This PR corrects that problem.

In an attempt to answer your question about PHPUnit and process isolation, I ran the following:

```bash
$ vendor/bin/phpunit --process-isolation tests/VCR/Util/StreamProcessorTest.php
PHPUnit 3.7.38 by Sebastian Bergmann.

Configuration read from /Users/mike/Code/php-vcr/php-vcr/phpunit.xml

..........

Time: 5.93 seconds, Memory: 4.25Mb
```

Hopefully that's enough to green-light this patch.  If there's more I can do to help verify this as working, please let me know.

**Comment by mcrumm** [comment]
D'oh, but of course PHPUnit didn't gripe about it.  PHP-VCR's dependencies don't share namespaces.  I'll try a more specific test and report back.

**Comment by mcrumm** [comment]
Well, I had to write a PHPUnit test for my project to be sure, but I can confirm that this works under process isolation for at least PHPUnit v3.7.38 (VCR's version from `composer.lock`) and PHPUnit v5.1.3.

**Comment by adri** [comment]
Alright, let's merge then.

**Comment by mcrumm** [comment]
Thanks!

**Comment by mdarse** [comment]
Thanks for merging this :+1:

**Pull request merged**